wherehows-web/app/services/configurator.ts

```diff
@@ -51,15 +51,21 @@ export default class Configurator extends Service {
    * @static
    * @template K
    * @param {K} [key] if provided, the value is returned with that key on the config hash is returned
+   * @param {IAppConfigOrProperty<K>} [defaultValue] if provided, will default if key is not found in config
    * @returns {IAppConfigOrProperty<K>}
    * @memberof Configurator
    */
-  static getConfig<K extends keyof IAppConfig | undefined>(key?: K): IAppConfigOrProperty<K> {
+  static getConfig<K extends keyof IAppConfig | undefined>(
+    key?: K,
+    defaultValue?: IAppConfigOrProperty<K>
+  ): IAppConfigOrProperty<K> {
     // Ensure that the application configuration has been successfully cached
     assert('Please ensure you have invoked the `load` method successfully prior to calling `getConfig`.', configLoaded);
 
     return typeof key === 'string' && appConfig.hasOwnProperty(<keyof IAppConfig>key)
       ? <IAppConfigOrProperty<K>>deepClone(appConfig[<keyof IAppConfig>key])
-      : <IAppConfigOrProperty<K>>deepClone(appConfig);
+      : defaultValue !== undefined
```

Comment:
Oh yeah, didn't think of that. I guess rather than arbitrarily looking for a defaultValue, we should have an interface that expresses desire to use a default value and then take whatever's there